### PR TITLE
[CAR_TESTS] Added carServiceTests

### PR DESCRIPTION
**Pull request by OlegTuryk**

### CODE REVIEW

**Review by Zel1oy — APPROVED** [review]
LEGITIME

File: src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Test
+    @DisplayName("Update car - Car exists")
+    public void update_CarExists_ReturnUpdatedCarDto() {
+        when(carMapper.toEntity(createCarRequestDto)).thenReturn(testCar);
+        when(carRepository.save(testCar)).thenReturn(testCar);
+        when(carMapper.toDto(testCar)).thenReturn(testCarDto);
+
+        CarDto actual = carService.update(1L, createCarRequestDto);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java]: Magic number

File: src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        when(carRepository.findById(nonExistentId)).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () -> carService.getById(nonExistentId));
+    }
+
+    @Test
+    @DisplayName("Delete car by id")
+    public void deleteById_CarExists_DeleteSuccessfully() {
+        doNothing().when(carRepository).deleteById(1L);
+        carService.deleteById(1L);
+        verify(carRepository, times(1)).deleteById(1L);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java]: Magic numbers

File: src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        List<CarDto> actual = carService.findAll(pageable);
+        assertEquals(List.of(testCarDto), actual);
+    }
+
+    @Test
+    @DisplayName("Get car by id - Car exists")
+    public void getById_CarExists_ReturnCarDto() {
+        when(carRepository.findById(1L)).thenReturn(Optional.of(testCar));
+        when(carMapper.toDto(testCar)).thenReturn(testCarDto);
+        CarDto actual = carService.getById(1L);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java]: Magic numbers

File: src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        assertNotNull(actual);
+        assertEquals(testCarDto, actual);
+    }
+
+    @Test
+    @DisplayName("Find all cars")
+    public void findAll_ReturnAllCars() {
+        Pageable pageable = PageRequest.of(1, 2);
+        Page<Car> carPage =
+                new PageImpl<>(List.of(testCar), pageable, 1);
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/service/CarServiceTest.java]: Magic numbers

File: src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        Assertions.assertEquals(expected.size(), actual.size());
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Test get endpoint for cars")
+    @WithMockUser(username = "CUSTOMER", roles = {"CUSTOMER"})
+    void getById_validCar_returnResponse() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/cars/1")
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java]: Magic number

File: src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        assertTrue(EqualsBuilder.reflectionEquals(carDto1, actual, "id"));
+    }
+
+    @Test
+    @WithMockUser(username = "MANAGER", roles = {"MANAGER"})
+    @DisplayName("Delete car by valid id")
+    void delete_validId_successful() throws Exception {
+        mockMvc.perform(delete("/cars/1"))
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java]: Magic number

File: src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+                .andExpect(status().isNoContent());
+    }
+
+    @Test
+    @WithMockUser(username = "MANAGER", roles = {"MANAGER"})
+    @DisplayName("Update car by id")
+    void update_validRequest_returnsExpectedCar() throws Exception {
+        createCarRequestDto1.setModel("Model Y");
+        carDto1.setModel("Model Y");
+        MvcResult mvcResult = mockMvc.perform(put("/cars/1")
```
Comment (padu12) [inline src/test/java/ua/mate/team3/carsharingapp/controller/CarControllerTest.java]: Magic number

**Review by padu12 — CHANGES_REQUESTED** [review]
Nice tests! Let's resolve all magic numbers. Please, pay more attention to magic numbers in future test

**Review by padu12 — APPROVED** [review]

**Review by SaiyaJ1N — APPROVED** [review]

**Pull request merged**